wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/jpa/JpaFactoryDao.java

```diff
@@ -159,7 +161,9 @@ protected FactoryImpl doUpdate(FactoryImpl update) throws NotFoundException {
         if (update.getWorkspace() != null) {
             update.getWorkspace().getProjects().forEach(ProjectConfigImpl::prePersistAttributes);
         }
-        return manager.merge(update);
+        FactoryImpl merged = manager.merge(update);
+        manager.flush();
```

Comment:
I can hardly imagine the listener which will create factory in some case :)) But for purposes of uniformity it's ok.